ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -638,6 +658,18 @@ private RuntimeBuildContext buildContext(
         StreamsConfig.METRIC_REPORTER_CLASSES_CONFIG,
         StorageUtilizationMetricsReporter.class.getName()
     );
+
+    // Passing shared state into managed components
+    newStreamsProperties.put(KsqlConfig.KSQL_INTERNAL_METRIC_COLLECTORS_CONFIG, metricCollectors);
+    newStreamsProperties.put(
+        KsqlConfig.KSQL_INTERNAL_METRICS_CONFIG,
+        metricCollectors.getMetrics()
+    );
+    newStreamsProperties.put(
+        KsqlConfig.KSQL_INTERNAL_STREAMS_ERROR_COLLECTOR_CONFIG,
+        new StreamsErrorCollector(applicationId, metricCollectors)
+    );
```

Comment:
Ok, here's where we're adding our shared objects to the Streams config. AK will pass them through to the Streams, Consumer, and Producer interceptors and handlers that need them.

Reply:
@tolgadur / @wcarlson5 , 

These are necessary to be able to start up Streams. I couldn't tell from the code in https://github.com/confluentinc/ksql/pull/8481 whether you're re-using the same config from the running app or building a new one from scratch. If it's the latter, then you're going to need this block as well. Which leads me to the question of why we would duplicate the contents of `buildStreamsProperties` to begin with instead of just calling it.

I didn't see any discussion of that point in #8481.

Reply:
We are rebuilding the config from scratch. I will open another PR to add it.